mutation_fragment_fwd.hh

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2016-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+#include <seastar/util/bool_class.hh>
+#include <seastar/util/optimized_optional.hh>
+
+using namespace seastar;
+
+namespace streamed_mutation {
+    // Determines whether streamed_mutation is in forwarding mode or not.
+    //
+    // In forwarding mode the stream does not return all fragments right away,
+    // but only those belonging to the current clustering range. Initially
+    // current range only covers the static row. The stream can be forwarded
+    // (even before end-of- stream) to a later range with fast_forward_to().
+    // Forwarding doesn't change initial restrictions of the stream, it can
+    // only be used to skip over data.
+    //
+    // Monotonicity of positions is preserved by forwarding. That is fragments
+    // emitted after forwarding will have greater positions than any fragments
+    // emitted before forwarding.
+    //
+    // For any range, all range tombstones relevant for that range which are
+    // present in the original stream will be emitted. Range tombstones
+    // emitted before forwarding which overlap with the new range are not
+    // necessarily re-emitted.
+    //
+    // When streamed_mutation is not in forwarding mode, fast_forward_to()
+    // cannot be used.
+    class forwarding_tag;
+    using forwarding = bool_class<forwarding_tag>;
+}
```

Comment:
I think that if we are moving this, it should be moved to the header which declares the flat mutation reader (`flat_mutation_reader.hh`). It doesn't really belong to `mutation_fragment.hh`.

Reply:
Also, orthogonal to this, the comment explaining something should be where the main declaration is, no one will think to look for the documentation in a `_fwd` header.